src/terms/ewald.jl

```diff
@@ -43,19 +61,20 @@ For now this function returns zero energy and force on non-3D systems. Use a pai
 potential term if you want to customise this treatment.
 """
 function energy_forces_ewald(lattice::AbstractArray{T}, charges, positions;
-                             η=nothing) where {T}
+                             η=default_η(lattice), q=zero(Vec3{T}), ph_disp=nothing) where {T}
+    S = isnothing(ph_disp) ? T : promote_type(complex(T), eltype(ph_disp[1]))
```

Comment:
This feels very hacky to me. Can't we solve this by a function barrier ? I.e. have one function `energy_forces_ewald(lattice, charges, positions, q, ph_disp)` which then calls a function `energy_forces_ewald(T, lattice::AbstractVector{T1}, charges, positions, q, ph_disp)` with `T = promote_type(complex(T1), eltype(ph_disp[1]))` as well as a function  `energy_forces_ewald(T, lattice::AbstractVector{T1}, charges, positions)`  which calls `energy_forces_ewald(T1, lattice::AbstractVector{T1}, charges, positions, zeros(T), nothing)`